### PR TITLE
#162725274 Highlight and comment on any text in an article

### PR DESCRIPTION
**Pull request by d-rita**
### What does this PR do?
This PR enables users to highlight and comment on any text in an article. 

#### Description of Task to be completed?
When a user is scrolling through an article and wants to comment on a particular part, he/she is able to select the desired text and comment about it. 

#### How should this be manually tested?
- Fetch this branch in your terminal and check out into it using `git fetch origin ft-highlight-and-comment-on-article-162725274 && git checkout ft-highlight-and-comment-on-article-162725274`
- To avoid conflicts; in the psql shell, drop the existing database using `drop database <database name>;` Then, create the database using `create database <database name>;` 
- Apply migrations by running `python manage.py migrate comments`
- Start the application by running the server with `python manage.py runserver`
- To highlight a particular text in an article for commenting, the request body will contain the start and ending index positions as well as the comment, i.e. `{
                                                   "body": "a comment body", 
                                                    "highlight_start": <integer value for the beginning of highlighted text>, 
                                                    "highlight_end": <integer value for the end of highlighted text> 
                                                }`
#### Screenshots
highlight

#### What are the relevant pivotal tracker stories?
[#162725274](https://www.pivotaltracker.com/story/show/162725274)

#### Postman Collection
[Run in Postman](https://app.getpostman.com/run-collection/77211431fb008ed18e9d)

### CODE REVIEW

**Review by MuhweziDeo — COMMENTED** [review]
@d-rita  isnt it supposed to be one migration cc @ja-odur

File: authors/apps/authentication/tests_/test_data.py [hunk 1]
```diff
@@ -168,3 +168,16 @@
...
+        'highlight_start':3, 
+        'highlight_end':10,
+        'body':'awesome'
+    }, 
+    'backward_highlight': {
+        'highlight_start':10, 
+        'highlight_end':3,
+        'body':'awesome'
+    }
+}
```
Comment (MuhweziDeo) [inline authors/apps/authentication/tests_/test_data.py]: add a space here @d-rita

**Review by MuhweziDeo — COMMENTED** [review]
Functionality wise everything is fine

**Review by MuhweziDeo — DISMISSED** [review]
LGTM

File: authors/apps/authentication/tests_/test_data.py [hunk 2]
```diff
@@ -1,31 +1,31 @@
 
 responses = {
-    'test_login_with_invalid_user_fails':{
+    'test_login_with_invalid_user_fails': {
```
Comment (ja-odur) [inline authors/apps/authentication/tests_/test_data.py]: @d-rita please avoid unnecessary changes

File: authors/apps/comments/migrations/0001_initial.py [hunk 3]
```diff
@@ -21,6 +21,9 @@ class Migration(migrations.Migration):
                 ('body', models.TextField(max_length=500)),
                 ('createdAt', models.DateTimeField(auto_now_add=True)),
                 ('updatedAt', models.DateTimeField(auto_now=True)),
+                ('highlight_start', models.IntegerField(null=True)),
```
Comment (ja-odur) [inline authors/apps/comments/migrations/0001_initial.py]: @d-rita dont merge migrations

File: authors/apps/authentication/tests_/test_data.py (hunk 2 above)
Comment (MuhweziDeo) [reply authors/apps/authentication/tests_/test_data.py]: @ believe this is a linter

File: authors/apps/comments/migrations/0001_initial.py (hunk 3 above)
Comment (MuhweziDeo) [reply authors/apps/comments/migrations/0001_initial.py]: @ja-odur  I thought we are supposed to push one migration file

File: authors/apps/authentication/tests_/test_data.py (hunk 2 above)
Comment (ja-odur) [reply authors/apps/authentication/tests_/test_data.py]: Why are we linting work that is not ours? PRs are when to capture only ones changes

**Review by ja-odur — CHANGES_REQUESTED** [review]
@d-rita PR should only reflect your changes

**Review by MuhweziDeo — COMMENTED** [review]
coverage has just dropped

File: authors/apps/comments/migrations/0001_initial.py (hunk 3 above)
Comment (ja-odur) [reply authors/apps/comments/migrations/0001_initial.py]: @d-rita One migrations for your changes (per feature or story), but does not mean you update existing migration versions.

Comment (MuhweziDeo) [reply authors/apps/comments/migrations/0001_initial.py]: makes sense 
@ja-odur  
@d-rita  had pushed her migration then I told her to delete it and make one migration for the whole thing my bad

File: authors/apps/authentication/tests_/test_data.py (hunk 2 above)
Comment (d-rita) [reply authors/apps/authentication/tests_/test_data.py]: noted. will revert

File: authors/apps/comments/migrations/0001_initial.py (hunk 3 above)
Comment (d-rita) [reply authors/apps/comments/migrations/0001_initial.py]: That is what i remembered as well. But i have worked on it.

**Comment by ja-odur** [comment]
@d-rita what happens when I put an `highlight-end`  greater than the body length?

**Comment by d-rita** [comment]
@ja-odur As per this code, it would return a null. I have validated against it though and it will be reflected in the pushed changes.

**Review by ja-odur — APPROVED** [review]
LGTM

**Review by MuhweziDeo — APPROVED** [review]

**Review by armstrongsouljah — APPROVED** [review]
LGTM!

**Review by mariamiah — APPROVED** [review]
LGTM

**Review by calvinpete — APPROVED** [review]
LGTM
